### PR TITLE
[hotfix] improvements to savepoint docs

### PR DESCRIPTION
**Pull request by knaufk**
## What is the purpose of the change

* minor improvements to savepoint docs

## Brief change log

see commits

### CODE REVIEW

File: docs/content/docs/ops/state/savepoints.md [hunk 1]
```diff
@@ -199,23 +200,20 @@ This submits a job and specifies a savepoint to resume from. You may give a path
...
-Flink will not delete its files, moreover, Flink can not depend on the existence of the files from
-such a snapshot, as it might be deleted outside of Flink's control. 
+As mentioned, the restore mode determines who takes over ownership of the files of the snapshots that we are restoring from. 
+Snapshots can be owned either by a user or Flink itself. 
+If a snapshot is owned by a user,  Flink will not delete its files, moreover, Flink can not depend on the existence of the files from such a snapshot, as it might be deleted outside of Flink's control. 
 
-There is no one perfect restore mode, each comes with its own price, but they can serve different
-purposes. Nevertheless, we believe the default *NO_CLAIM* mode is a good tradeoff for most usages,
-as it provides clear ownership with a small price for the first checkpoint after the restore.
+Each restore mode serves a specific purposes.
```
Comment (dawidwys) [inline docs/content/docs/ops/state/savepoints.md:215]: `purposes` -> `purpose`

File: docs/content/docs/ops/state/savepoints.md [hunk 2]
```diff
@@ -229,40 +227,31 @@ user's control and never delete any of the files. In this mode you can start mul
...
 <div style="text-align: center">
   {{< img src="/fig/restore-mode-no_claim.svg" alt="NO_CLAIM restore mode" width="70%" >}}
 </div>
 
 **CLAIM**
 
-The other available mode is the *CLAIM* mode. In this mode Flink will claim the ownership of the snapshot
-and might delete it at a certain point in time, once it becomes subsumed. Flink keeps around
+The other available mode is the *CLAIM* mode. In this mode Flink claims ownership of the snapshot
+and essentially treats it like a checkpoint: its controls the lifecycle and might delete it if it is not needed for recovery anymore.
```
Comment (dawidwys) [inline docs/content/docs/ops/state/savepoints.md:244]: `its controls` -> `it controls`? `controls its lifecycle`?

File: docs/content/docs/ops/state/savepoints.md [hunk 3]
```diff
@@ -229,40 +227,31 @@ user's control and never delete any of the files. In this mode you can start mul
...
-the snapshot, while others still depend on it.
-{{< /hint >}}
+of checkpoints.
 
 <div style="text-align: center">
   {{< img src="/fig/restore-mode-claim.svg" alt="CLAIM restore mode" width="70%" >}}
 </div>
 
-{{< hint warning >}}
+{{< hint info >}}
```
Comment (dawidwys) [inline docs/content/docs/ops/state/savepoints.md:254]: tbh, I'd rather keep it as a `warning` as those are limitations, but I am not strong on that

**Review by dawidwys — APPROVED** [review]
Thanks for the improvements. LGTM

**Comment by dawidwys** [comment]
One more comment. Just to say it out loud.

I think using contractions i.e. don't should be fine. After all, it is not that formal ;)

**Comment by dawidwys** [comment]
@knaufk shall we proceed with that?

**Comment by knaufk** [comment]
@dawidwys Thank you for the review. I addressed your comments. I once read that contractions should in general not be used in written language. I don't have a strong opinion.

**Pull request merged**